src/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion.py

```diff
@@ -496,11 +496,22 @@ def encode_image(self, image, device, num_images_per_prompt):
             image = self.feature_extractor(image, return_tensors="pt").pixel_values
 
         image = image.to(device=device, dtype=dtype)
-        image_embeds = self.image_encoder(image).image_embeds
-        image_embeds = image_embeds.repeat_interleave(num_images_per_prompt, dim=0)
+        if self.image_encoder.config.output_hidden_states:
```

Comment:
instead of changing the config for image_encoder let's:
1. add a argument `output_hidden_state` to `encode_image()`
2. inside the ip-adaper specific code in the pipelines (i.e. here https://github.com/huggingface/diffusers/blob/f72b28c75b2b4b720a5d8de78556694cf4b893fd/src/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion.py#L873), you can use the same logic you used before to determine the value of `output_hidden_state` and pass it to `encode_image()`, e.g. `output_hidden_state = False if isinstance(self.unet.encoder_hid_proj, ImageProjection) else True`

essentially we just want to keep `encode_image` more generic and put ip-adapter specific log into the `if ip_adapter_image is not None:  ....`